templates/k8s.yaml

```diff
@@ -95,10 +95,18 @@ provision:
   script: |
     #!/bin/bash
     set -eux -o pipefail
-    test -e /etc/kubernetes/admin.conf && exit 0
-    export KUBECONFIG=/etc/kubernetes/admin.conf
+    test -e /run/lima-k8s-pull-done && exit 0
```

Comment:
Not a fan of introducing more `/run/lima-*` files.
Can we rather just check `kubeadm config images list` ?

Reply:
I can move it to some other location perhaps, outside `/run`.
The "list" command only prints, there is no "check" unfortunately.

Reply:
I had some other files in /etc/kubernetes/tmp, so used that instead.

Reply:
> No `nerdctl image exists` either (like podman), but there is `inspect` The list command is somewhat expensive as it checks online though.

Untested idea: You could `nerdctl image tag` it with an obscure tag name. Just verify that it continues to fail until the image is fully downloaded.

Reply:
Ok, confirmed that it fails fast with status 1 while the download is still in progress:

```console
❯ nerdctl image tag python:latest python:foo
FATA[0000] python:latest: not found
Error: exit status 1
```

And then has no output and exit status 0 once it succeeds. And you can run it multiple times once it succeeds, and it continues to succeed (doesn't complain that the tag already exists).

Reply:
I added a feature for it to `kubeadm`, but they liked the current situation better

https://github.com/kubernetes/kubernetes/commit/8433cc602f91c54cc52c89f29c6013891c2a1254

It could have been easily exposed in the `kubeadm config images` subcommands.

```
Usage:
  kubeadm config images [flags]
  kubeadm config images [command]

Available Commands:
  list        Print a list of images kubeadm will use. The configuration file is used in case any images or image repositories are customized
  pull        Pull images used by kubeadm
```

Reply:
> Untested idea: You could `nerdctl image tag` it with an obscure tag name.

My normal workaround is just $(DOCKER) image inspect $(IMAGE) >/dev/null

Reply:
My comment was not about the path of the file; I wonder if the file could be just removed

Reply:
But you mean something like this?

`for image in $(kubeadm config images list); do nerdctl -n k8s.io inspect $image >/dev/null; done`